### ISSUE TITLE
Space Supporter finish line: Refactor

### ISSUE THREAD

**Issue by monamohebbi** [cloudfoundry/cloud_controller_ng#2444]
## Issue

We've completed the work for the Space Supporter role! Now let's make sure our the cloud controllers permissions logic is still readable and sustainable.

## Context

As we have been developing this role incrementally, we have been moving forward with some patterns that we always knew would be temporary until we were in a steady and finished state. Now that we are (hopefully) in that state, it is time to do some cleaning up and settle on the names and patterns we hope to move forward with.

## Places to Refactor

- [x] Replace all instances of shared example tests that send in space supporter permissions as an [extra addition](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/spec/request/apps_spec.rb#L555), and instead include the space supporter in [ALL_PERMISSIONS](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/spec/request_spec_shared_examples.rb#L7-L15). Fiddle with tests until they are all passing, or submit an issue if this brings any bugs or missing endpoints to the surface. 
- [x] Now that the space supporter has access to most read endpoints, let's rename [ROLES_FOR_SPACE_SUPPORTER_READING](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L24-L30) to ROLES_FOR_READING, and rename the old ROLES_FOR_READING to something more appropriate regarding the few secret endpoints that still remain off limits to the supporter. Do a similar rename for the [functions](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L176-L182) that make use of these permissions (let's get rid of the untrusted name!).
- [x] Rename [ROLES_FOR_SPACE_SUPPORTER_WRITING](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L66-L68) keeping in mind that currently there are already two functions for space writing [here](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L53-L55) and [here](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L62-L64) that map to the same permissions. Do a similar rename for the [functions](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L193-L207) that make use of these permissions (let's get rid of the untrusted name!).
- [x] Include the space supporter in [SPACE_ROLES](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L43-L45). Rename [functions](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L250-L264) that make use of space roles following the patterns from the previous refactors.

## Expected result

All unit tests still pass.

## Help from the community:
If you've worked on this track and can think of other areas that might be left in an undesirable state, comment your thoughts below!

**Comment by ctlong** [comment]
`ROLES_FOR_SPACE_SERVICES_READING` could probably use some consideration. We broke from the previous pattern because we didn't feel it made sense to have a constant titled `ROLES_FOR_SPACE_SECRETS_READING_INCLUDING_SUPPORTER`. Is this the correct name to use though?

**Comment by philippthun** [comment]
Not sure which constants will remain in [permissions.rb](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/lib/cloud_controller/permissions.rb), but when extending a list, it would be good not to duplicate the list, but to stick to the following pattern: https://github.com/cloudfoundry/cloud_controller_ng/blob/92e3ae6fecbaaa286d359bdcf733c19f4533510f/lib/cloud_controller/permissions.rb#L43-L45

**Comment by ctlong** [comment]
Does this story include taking a look through the docs to ensure that they're consistent?

**Comment by monamohebbi** [comment]
@ctlong 
> ROLES_FOR_SPACE_SERVICES_READING could probably use some consideration

Agreed. As it stands now we're technically not in a 'dry' state here because as it stands `ROLES_FOR_SPACE_SERVICES_READING` and `UNTRUSTED_CAN_WRITE_TO_SPACE` point to the same function. But I'm not sure there's any clear way to merge these two with out having a very confusing method call somewhere else. I think it might be for the best to leave the duplication to leave the code in a more readable state as it's only a 2 line duplication. This is not the first time we've had to have a little [duplication](https://github.com/cloudfoundry/cloud_controller_ng/blob/fc0955f790f63705bdd1d39cd7f2d088b37a1958/lib/cloud_controller/permissions.rb#L9-L15) in the permissions divisions either.

> Does this story include taking a look through the docs to ensure that they're consistent?

To the best of our ability! But I'm sure the community will still find some inconsistencies down the road and we encourage anyone to call out any errors we might have made.

@philippthun 
I agree! We made sure not to get rid of any place that introduced it.
